rascal-lsp/src/main/rascal/library/util/LanguageServer.rsc

```diff
@@ -270,8 +270,17 @@ data LanguageService
     | implementation(set[loc] (Focus _focus) implementationService)
     | codeAction    (list[CodeAction] (Focus _focus) codeActionService)
     | selectionRange(list[loc](Focus _focus) selectionRangeService)
+    | formatting    (list[TextEdit](Tree _input, loc range, FormattingOptions _opts) formattingService)
```

Comment:
I like that `range` is a parameter. Please also add a documentation section that introduces how formatting works and how this range parameter should be interpreted (like when is it the whole file? if it's equal to `.top`?)